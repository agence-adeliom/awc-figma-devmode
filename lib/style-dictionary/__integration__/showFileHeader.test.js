/*
 * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

const { fs } = require("../lib/utils/store");
const helper = require("../__tests__/__helpers");
const StyleDictionary = require("../index");
const { buildPath } = require("./_constants");

describe("integration", () => {
  describe("showFileHeader", () => {
    StyleDictionary.extend({
      // we are only testing showFileHeader options so we don't need
      // the full source.
      source: helper.getTokens([`./__integration__/tokens/size/padding.json`]),
      platforms: {
        css: {
          transformGroup: "css",
          buildPath,
          files: [
            {
              destination: "platform-none-file-none.css",
              format: "css/variables",
            },
            {
              destination: "platform-none-file-false.css",
              format: "css/variables",
              options: {
                showFileHeader: false,
              },
            },
          ],
        },
        fileHeader: {
          transformGroup: "css",
          buildPath,
          options: {
            showFileHeader: false,
          },
          files: [
            {
              destination: "platform-false-file-none.css",
              format: "css/variables",
            },
            {
              destination: "platform-false-file-true.css",
              format: "css/variables",
              options: {
                showFileHeader: true,
              },
            },
          ],
        },
      },
    }).buildAllPlatforms();

    describe(`without platform options`, () => {
      it(`should show file header if no file options set`, () => {
        const output = fs.get(`${buildPath}platform-none-file-none.css`);
        expect(output).toMatchSnapshot();
      });

      it(`should not show file header if file options set to false`, () => {
        const output = fs.get(`${buildPath}platform-none-file-false.css`);
        expect(output).toMatchSnapshot();
      });
    });

    describe(`with platform options set to false`, () => {
      it(`should not show file header if no file options set`, () => {
        const output = fs.get(`${buildPath}platform-false-file-none.css`);
        expect(output).toMatchSnapshot();
      });

      it(`should show file header if file options set to true`, () => {
        const output = fs.get(`${buildPath}platform-false-file-true.css`);
        expect(output).toMatchSnapshot();
      });
    });
  });
});

afterAll(() => {
  fs.rm(buildPath);
});
