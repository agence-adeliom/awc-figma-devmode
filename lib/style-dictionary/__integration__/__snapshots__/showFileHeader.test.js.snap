// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration showFileHeader with platform options set to false should not show file header if no file options set 1`] = `
":root {
  --size-padding-small: 0.5rem;
  --size-padding-medium: 1rem;
  --size-padding-large: 1rem;
  --size-padding-xl: 1rem;
}
"
`;

exports[`integration showFileHeader with platform options set to false should show file header if file options set to true 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

:root {
  --size-padding-small: 0.5rem;
  --size-padding-medium: 1rem;
  --size-padding-large: 1rem;
  --size-padding-xl: 1rem;
}
"
`;

exports[`integration showFileHeader without platform options should not show file header if file options set to false 1`] = `
":root {
  --size-padding-small: 0.5rem;
  --size-padding-medium: 1rem;
  --size-padding-large: 1rem;
  --size-padding-xl: 1rem;
}
"
`;

exports[`integration showFileHeader without platform options should show file header if no file options set 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

:root {
  --size-padding-small: 0.5rem;
  --size-padding-medium: 1rem;
  --size-padding-large: 1rem;
  --size-padding-xl: 1rem;
}
"
`;
